server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1531,7 +1533,10 @@ public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext
             }
 
             if ((operation == FielddataOperation.SEARCH || operation == FielddataOperation.SCRIPT) && hasDocValues()) {
-                return type.getFieldDataBuilder(name());
+                ValuesSourceType valuesSourceType = metricType == TimeSeriesParams.MetricType.counter
+                    ? TimeSeriesValuesSourceType.COUNTER
+                    : type.numericType.getValuesSourceType();
+                return type.getFieldDataBuilder(name(), valuesSourceType);
             }
 
             if (operation == FielddataOperation.SCRIPT) {
```

Comment:
Does the "script" version of this need `metricType` too?

Reply:
I guess it doesn't read the ValuesSourceType, but maybe it'd make sense to pass it along just in case? I suppose this is an OO-smell because scripts receive a thing they don't use.... But if we fixed all of those we'd go crazy.

Reply:
The script versions are all loading from source, so no field data involved.  It calls a separate method so I don't think we need to add the values source type here?

Reply:
Yeah, I wonder if we should rename the SEARCH context to AGGS, as that's generally why it's being loaded?

Reply:
I think it won't break anything not to pass the COUNTER to the SCRIPT type fields - but that's because they throw out the values source type when they pass it to scripts.

Reply:
I was thinking we should attach the metricType to that FieldData as well. I get that it isn't used, but if we don't attach the actual metric type we'll be attaching the wrong thing? Maybe they could return "null" for "I don't know"?

Reply:
oh I see what you mean, at the moment they get the value from the NumericType but that is indeed incorrect.  I'll update!